controllers/vsl.go

```diff
@@ -122,41 +122,38 @@ func (r *DPAReconciler) ValidateVolumeSnapshotLocations(dpa oadpv1alpha1.DataPro
 			Spec: *vslSpec.Velero,
 		}
 
+		// check if VSL name is specified in DPA spec
+		if vslSpec.Name != "" {
+			vsl.Name = vslSpec.Name
+		}
+
 		// check for valid provider
 		if vslSpec.Velero.Provider != AWSProvider && vslSpec.Velero.Provider != GCPProvider &&
 			vslSpec.Velero.Provider != Azure {
 			r.Log.Info("Non-supported provider specified, might be a misconfiguration")
 
-			r.EventRecorder.Event(&vsl,
-				corev1.EventTypeWarning,
-				"VSL provider is invalid",
-				fmt.Sprintf("VSL provider %s is invalid, might be a misconfiguration", vslSpec.Velero.Provider),
-			)
+			return false, fmt.Errorf("invalid provider %s for VSL %s", vslSpec.Velero.Provider, vsl.Name)
```

Comment:
I really think we should start to give more verbose error messages to users, as I explain here https://github.com/openshift/oadp-operator/issues/1536

My suggestion for here, for example, would be
> DPA spec.snapshotLocations[**index**].velero.provider is invalid: **wrong-one** is not a valid value. Use of of **options**

Reply:
Is the name of VSL needed then?

Reply:
I think just yaml path is fine (and most of the time users do not even put name on those objects, right? so it will not be a valuable information for them) 

With yaml path I would say error message is good enough, because if user does not understand it, we will understand it immediately and can easily help the user fix the problem

Reply:
ya sounds good. Thanks